src/EntityListener/ArticleListener.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace AppBundle\EntityListener;
+
+use AppBundle\Entity\Article;
+use AppBundle\Redirection\Dynamic\RedirectionManager;
+use Doctrine\ORM\Event\PreUpdateEventArgs;
+use Symfony\Bundle\FrameworkBundle\Routing\Router;
+
+class ArticleListener
+{
+    private $redirectionManager;
+    private $router;
+
+    /**
+     * @var array[]
+     */
+    private $redirections = [];
+
+    public function __construct(RedirectionManager $redirectionManager, Router $router)
+    {
+        $this->redirectionManager = $redirectionManager;
+        $this->router = $router;
+    }
+
+    public function preUpdate(Article $article, PreUpdateEventArgs $preUpdateEventArgs): void
+    {
+        if (!$article->isPublished()
```

Comment:
if the article isn't published you'll creat one redirect entry wich the same URL for `source` and `target`, why?